app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -927,13 +921,30 @@ protected function getRatesDebuggingData($rates) {
      */
     public function removePreAuthOrder($order) {
         if ($this->isBoltOrder($order)) {
+            ##########################################################
+            # Unblock quote to allow order to be re-processed
+            ##########################################################
+            $parentQuote = $this->getParentQuoteFromOrder($order);
+            $parentQuote->setIsActive(true)->save();
+            ###########################################################
+
+            ##############################################################
+            # Cancel order for restocking inventory and triggering events
+            ##############################################################
             if ($order->getStatus() !== 'canceled_bolt') {
                 $order->cancel()->setQuoteId(null)->setStatus('canceled_bolt')->save();
```

Comment:
@mccottry should we move the unblock quote logic to after the order is deleted from the system, otherwise, it might cause some unexpected issues (e.g. sending abandon cart email to the customer, etc).

Reply:
@johnmccombs1 I'm a little hesitant to change the order because in testing, I observed order was important -- though logically, I can not see why this would be the case.  I'll run a few test with the modified order that you suggest because the previous order should be gone before the new order is created, however, I also do not see why canceling or deleting an order would trigger an abandoned cart email.  Technically, those orders are fulfilled.  Have you observed this problem?